### ISSUE TITLE
Set limit/offset in search (not in paginate)

### ISSUE THREAD

**Issue by 2618094** [Jeroen-G/Explorer#200]
Hello!
I'm stuck in problem - `offset` value can be set only with `pagination` method. 

But in my case, i have my own pagination and need to set offset/limit in `search` method - but this is impossible.

For now, i solve this with redefinition of `ElasticEngine`. Search command look like
```
    public function search(Builder $builder): Results
    {
        $normalizedBuilder = ScoutSearchCommandBuilder::wrap($builder);
        if($offset = $builder->options['offset'] ?? null){
            $normalizedBuilder->setOffset($offset);
        }
        if($limit = $builder->options['limit'] ?? null){
            $normalizedBuilder->setLimit($limit);
        }
        self::$lastQuery = $normalizedBuilder->buildQuery();
        return $this->documentAdapter->search($normalizedBuilder);
    }
```

But this is **DISGUSTING**

I have two points about this problem:
1. Use this `options` array in `ScoutSearchCommandBuilder` like in my `search` implementation
2. Pass query in `Builder::callback` before perform search.

What you think about that? I can write this things and make PR, but need opinion about this variants (or a new look at this problem).

Thanks!

**Comment by Jeroen-G** [comment]
Hi there!

What you do mean with "_offset value can be set only with pagination method_"?

**Comment by 2618094** [comment]
The `ElasticEngine::paginate()` method has an `offset` and `limit` query params:
```
    public function paginate(Builder $builder, $perPage, $page): Results
    {
        $offset = $perPage * ($page - 1);

        $normalizedBuilder = ScoutSearchCommandBuilder::wrap($builder);
        $normalizedBuilder->setOffset($offset);
        $normalizedBuilder->setLimit($perPage);
        self::$lastQuery = $normalizedBuilder->buildQuery();
        return $this->documentAdapter->search($normalizedBuilder);
    }
```
However, the Scout result of this method is a `LenghtAwarePaginator`, not a `Builder`, which doesn't work for me.

I would like to be able to set `offset` and `limit` elastic query params in `ElasticEngine::search()` method like:
```
Model::search($query)->options([
            'limit'  => $limit,
            'offset' => $offset,
        ]);
```
Or like:
```
Model::search($query, function(ScoutSearchCommandBuilder $builder) use ($limit, $offset){
    $builder->setOffset($offset);
    $builder->setLimit($limit)
})
```

**Comment by 2618094** [comment]
Yes, scout support both of this options - callback in `search` and `options` method like i show.

But in this package both of them not used for now - and both of them can be used for this purporses.

**Comment by Jeroen-G** [comment]
I only see the callback mentioned in the docs, so let's go with that!

**Comment by Jeroen-G** [comment]
Fixed in #201

**Comment by Jeroen-G** [comment]
Released in 3.7.0
